appservice/src/main/java/nu/yona/server/subscriptions/rest/UserController.java

```diff
@@ -146,27 +147,50 @@
 		return createOkResponse(userService.getPublicUser(userId), false);
 	}
 
-	@RequestMapping(value = "/{userId}/apple.mobileconfig", method = RequestMethod.GET)
+	@RequestMapping(value = "/{userId}/apple.mobileconfig", params = { "sign" }, method = RequestMethod.GET)
 	@ResponseBody
 	public ResponseEntity<byte[]> getAppleMobileConfig(
-			@RequestHeader(value = Constants.PASSWORD_HEADER) Optional<String> password, @PathVariable UUID userId)
+			@RequestHeader(value = Constants.PASSWORD_HEADER) Optional<String> password,
+			@RequestParam(value = "sign", defaultValue = "false") String signStr, @PathVariable UUID userId)
 	{
 		HttpHeaders headers = new HttpHeaders();
 		headers.setContentType(new MediaType("application", "x-apple-aspen-config"));
+		boolean mustSign = Boolean.TRUE.toString().equals(signStr);
 		try (CryptoSession cryptoSession = CryptoSession.start(password, () -> userService.canAccessPrivateData(userId)))
 		{
-			return new ResponseEntity<>(getUserSpecificAppleMobileConfig(userService.getPrivateUser(userId)), headers,
+			return new ResponseEntity<>(getUserSpecificAppleMobileConfig(userService.getPrivateUser(userId), mustSign), headers,
 					HttpStatus.OK);
 		}
 	}
 
-	private byte[] getUserSpecificAppleMobileConfig(UserDto privateUser)
+	@RequestMapping(value = "/{userId}/apple.mobileconfig", method = RequestMethod.GET)
+	@ResponseBody
+	public ResponseEntity<byte[]> getAppleMobileConfig(
+			@RequestHeader(value = Constants.PASSWORD_HEADER) Optional<String> password, @PathVariable UUID userId)
+	{
+		return getAppleMobileConfig(password, Boolean.FALSE.toString(), userId);
+	}
+
+	private byte[] getUserSpecificAppleMobileConfig(UserDto privateUser, boolean mustSign)
 	{
 		Context ctx = ThymeleafUtil.createContext();
 		ctx.setVariable("ldapUsername", privateUser.getPrivateData().getVpnProfile().getVpnLoginId().toString());
 		ctx.setVariable("ldapPassword", privateUser.getPrivateData().getVpnProfile().getVpnPassword());
 
-		return templateEngine.process("apple.mobileconfig.xml", ctx).getBytes(StandardCharsets.UTF_8);
+		return signIfRequested(mustSign, templateEngine.process("apple.mobileconfig.xml", ctx).getBytes(StandardCharsets.UTF_8));
+	}
+
+	private byte[] signIfRequested(boolean mustSign, byte[] unsignedMobileconfig)
+	{
+		if (mustSign)
+		{
+			AppleMobileConfigSigningProperties properties = yonaProperties.getSecurity().getAppleMobileConfigSigning();
+			String signingCertificateFile = properties.getSigningCertificateFile();
+			String signingKeyFile = properties.getSigningKeyFile();
+			String password = properties.getPassword();
+			return new Signer(signingCertificateFile, signingKeyFile, password).sign(unsignedMobileconfig);
```

Comment:
Qua performance it would be good to cache the instance of the `Signer` with the certificate/key files preloaded. Can't this be done with `@Autowired`?